File: app/src/main/java/com/skydoves/waterdays/compose/qualifiers/RequirePresenter.java
package com.skydoves.waterdays.compose.qualifiers;

import com.skydoves.waterdays.compose.BasePresenter;

import java.lang.annotation.Inherited;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * Developed by skydoves on 2017-08-19.
 * Copyright (c) 2017 skydoves rights reserved.
 */

@Inherited
@Retention(RetentionPolicy.RUNTIME)
public @interface RequirePresenter {
  Class<? extends BasePresenter> value();
}

File: app/src/main/java/com/skydoves/waterdays/compose/qualifiers/RequirePresenter.kt
package com.skydoves.waterdays.compose.qualifiers

import com.skydoves.waterdays.compose.BasePresenter
import java.lang.annotation.Inherited
import kotlin.reflect.KClass

/**
 * Developed by skydoves on 2017-08-19.
 * Copyright (c) 2017 skydoves rights reserved.
 */

@Inherited
@Retention(AnnotationRetention.RUNTIME)
annotation class RequirePresenter(val value: KClass<out BasePresenter<*>>)
